src/UnigramWordPredictor.java

```diff
@@ -24,88 +25,145 @@ public UnigramWordPredictor(Tokenizer tokenizer) {
 
   /**
    * Trains the predictor using the text provided by the Scanner.
-   * The method tokenizes the text and builds a map where each word 
-   * is associated with a list of words that immediately follow it 
+   * The method tokenizes the text and builds a map where each word
+   * is associated with a list of words that immediately follow it
    * in the text. The resultant map is stored in the neighborMap
    * instance variable.
    * 
    * For example:
    * If the input text is: "The cat sat. The cat slept. The dog barked."
-   * After tokenizing, the tokens would be: ["the", "cat", "sat", ".", "the", "cat", "slept", ".", "the", "dog", "barked", "."]
+   * After tokenizing, the tokens would be: ["the", "cat", "sat", ".", "the",
+   * "cat", "slept", ".", "the", "dog", "barked", "."]
    * 
    * The resulting map (neighborMap) would be:
    * {
-   *   "the" -> ["cat", "cat", "dog"],
-   *   "cat" -> ["sat", "slept"],
-   *   "sat" -> ["."],
-   *   "." -> ["the", "the"],
-   *   "slept" -> ["."],
-   *   "dog" -> ["barked"],
-   *   "barked" -> ["."]
+   * "the" -> ["cat", "cat", "dog"],
+   * "cat" -> ["sat", "slept"],
+   * "sat" -> ["."],
+   * "." -> ["the", "the"],
+   * "slept" -> ["."],
+   * "dog" -> ["barked"],
+   * "barked" -> ["."]
    * }
    * 
    * The order of the map and the order of each list is not important.
    * 
    * @param scanner the Scanner to read the training text from
    */
+
   public void train(Scanner scanner) {
+
+    // List of words containing input from scanner
     List<String> trainingWords = tokenizer.tokenize(scanner);
+    // Create empty hashmap with String as keys and a list of strings as value
+    Map<String, List<String>> neighborMap = new HashMap<>();
```

Comment:
This works, but you could consider just using the instance variable directly instead of creating a new instance variable and reassigning the instance variable at the end of your method.